odoo/addons/base/models/ir_module.py

```diff
@@ -462,10 +462,16 @@ def button_immediate_install(self):
         return self._button_immediate_function(self.env.registry[self._name].button_install)
 
     @assert_log_admin_access
-    def button_install_cancel(self):
-        self.write({'state': 'uninstalled', 'demo': False})
+    @api.model
+    def button_reset_state(self):
+        self.search([('state', '=', 'to install')]).state = 'uninstalled'
+        self.search([('state', 'in', ('to update', 'to remove'))]).state = 'installed'
```

Comment:
The diff makes this a bit confusing, but this **new** method cleans up the states of **all** modules. :+1: